jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -352,9 +352,10 @@ public boolean isAlias()
     }
 
     /**
-     * The canonical Alias for the Resource as a URI.
+     * If this Resource is an alias pointing to a different location,
+     * the resulting location as URI is returned pointing to the real location.
      *
-     * @return The canonical Alias of this resource or null if none.
+     * @return The real URI location of this resource or null if not different.
      */
     public URI getAlias()
```

Comment:
The name of this method is confusing, because it does not return an alias, but the real location.

Where this method is used, the variable name is typically `realURI` so perhaps it should be renamed to `getResolvedRealURI()`, with the `Resolved` part indicating that if there was no resolution because the resource was already a real location, then `null` is returned.

I could also live with `getRealURI()` always return the real URI and for the 3-4 places this method is used rather than a null check do an equal check: `if (!resource.getURI().equals(resource.getRealURI()))`, or equivalently a call to `isAlias()`.
I like this latter solution better because it will be evident what you're testing for, rather than doing a null check and having to read the javadocs to understand that `null` means that the resource URI is already a real URI.

Reply:
I don't like the name "get*RealURI()" as it conflicts with the `getURI()`, and has a name that indicates it will always return a value.
It _only returns a value_ only if there is a difference from what `getURI()` would return.

It's almost a `getAliasLocationURI()`, but it still stems from how backwards this API is.

Currently, it is .. Alias is `this`, and the `.getAlias()` shows the real file location / URI.

Reply:
Wonder if we can use the language from symlinks?

```
$ man ln

NAME
       ln - make links between files

SYNOPSIS
       ln [OPTION]... [-T] TARGET LINK_NAME
       ln [OPTION]... TARGET
       ln [OPTION]... TARGET... DIRECTORY
       ln [OPTION]... -t DIRECTORY TARGET...

DESCRIPTION
       In  the  1st form, create a link to TARGET with the name LINK_NAME.  In
       the 2nd form, create a link to TARGET in the current directory.  In the
       3rd  and  4th  forms, create links to each TARGET in DIRECTORY.  Create
       hard links by default, symbolic links  with  --symbolic.   By  default,
       each  destination  (name  of  new link) should not already exist.  When
       creating hard links, each TARGET must exist.  Symbolic links  can  hold
       arbitrary  text;  if  later resolved, a relative link is interpreted in
       relation to its parent directory.
```

In this language, the `PathResource` itself is the `LINK_NAME` and the `.getAlias()` is the `TARGET`.

How about `.getTargetURI()` ?

Reply:
Changed to `.getTargetURI()` (it reads better too)